cmd/clusterctl/clusterdeployer/bootstrap/kind/kind_test.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kind
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/pkg/errors"
+)
+
+func TestCreate(t *testing.T) {
+	var testcases = []struct {
+		name      string
+		execError error
+		expectErr bool
+	}{
+		{
+			name: "success",
+		},
+		{
+			name:      "exec fail",
+			execError: errors.New("test error"),
+			expectErr: true,
+		},
+	}
+	for _, testcase := range testcases {
+		t.Run(testcase.name, func(t *testing.T) {
+			m := New()
+			m.execFunc = func(args ...string) (string, error) {
+				return "", testcase.execError
+			}
+			err := m.Create()
+			if (testcase.expectErr && err == nil) || (!testcase.expectErr && err != nil) {
+				t.Fatalf("unexpected returned error, got: %v, want err: %v", err, testcase.expectErr)
+			}
+		})
+	}
+}
+
+func TestDelete(t *testing.T) {
+	var testcases = []struct {
+		name      string
+		execError error
+		expectErr bool
+	}{
+		{
+			name: "success",
+		},
+		{
+			name:      "exec fail",
+			execError: errors.New("test error"),
+			expectErr: true,
+		},
+	}
+	for _, testcase := range testcases {
+		t.Run(testcase.name, func(t *testing.T) {
+			m := New()
+			m.execFunc = func(args ...string) (string, error) {
+				return "", testcase.execError
+			}
+			err := m.Delete()
+			if (testcase.expectErr && err == nil) || (!testcase.expectErr && err != nil) {
+				t.Fatalf("unexpected returned error. Got: %v, Want Err: %v", err, testcase.expectErr)
+			}
+		})
+	}
+}
+
+func TestGetKubeconfig(t *testing.T) {
+	const contents = "dfserfafaew"
+	m := New()
+	f, err := createTempFile(contents)
+	if err != nil {
+		t.Fatal("unable to create test file")
```

Comment:
Strings passed to t.Fatal should actually be capitalized.

Reply:
From https://github.com/golang/go/wiki/CodeReviewComments#error-strings:

"Error strings should not be capitalized .... This does not apply to logging, which is implicitly line-oriented and not combined inside other messages."

It's that last sentence that applies here.